File: src/internal/mods/request/article.go
package request

type CreateArticleRequest struct {
	Title   string `json:"title"`
	Content string `json:"content"`
	Summary string `json:"summary"`
	Cover   string `json:"cover"`
}

type UpdateArticleRequest struct {
	ID      uint    `json:"id"`                // 文章 ID
	Title   *string `json:"title,omitempty"`   // 文章标题
	Content *string `json:"content,omitempty"` // 正文内容
	Summary *string `json:"summary,omitempty"` // 文章摘要
	Cover   *string `json:"cover,omitempty"`   // 封面图 URL
	IsTop   *bool   `json:"is_top,omitempty"`  // 是否置顶
}

type ArticleQueryListReq struct {
	Title    string `json:"title"`
	UserName string `json:"userName"`
	Summary  string `json:"summary"`
	PageReq
}

type UserArticleListReq struct {
	UserID uint `json:"userId"` // 用户 ID
	PageReq
}

File: web/src/api/article.ts
import { http } from './request'
import type { ArticleInfo, ArticleListItem, PageData } from '@/types'

/** 获取公开文章列表 */
export function getArticleList(params: {
  title?: string
  userName?: string
  summary?: string
  page?: number
  pageSize?: number
}) {
  return http.post<PageData<ArticleListItem>>('/pub/article/list', params)
}

/** 获取文章详情 */
export function getArticleInfo(id: number) {
  return http.get<ArticleInfo>('/pub/article/get_info', { id })
}

/** 获取指定用户的文章列表 */
export function getUserArticleList(userId: number, page = 1, pageSize = 10) {
  return http.post<PageData<ArticleListItem>>('/pub/article/list_by_user', {
    userId,
    page,
    pageSize,
  })
}

/** 创建文章（需登录） */
export function createArticle(data: {
  title: string
  content: string
  summary?: string
  cover?: string
}) {
  return http.post<string>('/pri/article/create', data)
}

/** 更新文章（需登录） */
export function updateArticle(data: {
  id: number
  title?: string
  content?: string
  summary?: string
  cover?: string
  isTop?: boolean
}) {
  return http.put<string>('/pri/article/update', data)
}

/** 删除文章（需登录） */
export function deleteArticle(id: number) {
  return http.delete<string>('/pri/article/delete', { id })
}
